### DISCUSSION TITLE
Absolute Time Range Is Not aligned to the Visibility data in Time Series with Infinity(Json) Data Source

### DISCUSSION THREAD

**Discussion by Umesh2214906** [grafana/grafana-infinity-datasource#550] [Q&A] [answered]
Hi Team,

I am Facing some issue in the Infinity(Json) Data Source.

We are facing an issue applying the time filter on the widgets could you please help us on this issue please.

Please find the below points to understand the issue more clear.

we are trying to apply the Time Filter to the widgets when we change the Absolute Time range the data in the widgets should also change please help us how to apply to the widgets.

•	We are building the dashboard  using via API.
o	Data Source Plugins
o            JSON API.
o            Infinity.
•	The Time series graph is working properly based on the Absolute Time Range, But for Rest of the widgets it is not working even though I have give date time in the Query.
•	We have tried the Macros Time filter but it is not working properly.
•	These are By default didn’t come when we installed the Grafana. These JSON API and Infinity we have downloaded from the Grafana Plugins Page.

Thanks in advance.

**Comment by yesoreyeram** [comment]
You need to properly pass the time range to your API url in such format that it accepts. Use the proper macro specified [here](https://grafana.com/docs/grafana/v8.5/variables/variable-types/global-variables/#__from-and-__to). 

In the query inspector, you can validate the computed url. ( if that doesn't work, check manually the URL results if they are time filtered via curl ).

**Comment by Umesh2214906** [comment]
Hi Sir,

Where exactly i have apply the Time Filter of the below lines

Grafana has two built in time range variables: $__from and $__to. They are currently always interpolated as epoch milliseconds by default but you can control date formatting.

in the Data Source i find the "Parsing options & Result fieldsField types, alias and selectors" & "Results Filter"

And Next there are Transforms.

Where exactly can i apply the Time filter to the data source.

Could you please help me on that.

Thanks in Advance.

**Reply by yesoreyeram** [reply]
either in the url or body.. depends on your api.

**Comment by Umesh2214906** [comment]
Hi Sir,

Could you please let me know which time of the Data Time format will be good to solve my issue.

Please suggest

Thanks in Advance.

**Reply by yesoreyeram** [reply]
What API you are trying to connect?

**Comment by Umesh2214906** [comment]
Hi Sir,

Can you please provide your Official email id, so that we can schedule a call with you and explain our problem.

From that you can understand our issue where we are facing the challenge that may help us a lot.

Thanks in Advance.

**Reply by yesoreyeram** [reply]
Sorry. I can't provide direct support in call.

**Comment by Umesh2214906** [comment]
Hi Sir,

Could you help us how can we contact support for our issue please.

Thanks in Advance.

**Comment by Umesh2214906** [comment]


**Reply by yesoreyeram** [reply]
use something like below..

```json
{
  "from_date" : "${__from:date:YYYY-MM-DD hh}:${__from:date:mm}:${__from:date:ss}",
  "to_date": "${__from:date:YYYY-MM-DD hh}:${__from:date:mm}:${__from:date:ss}"
}
```

<img width="825" alt="image" src="https://user-images.githubusercontent.com/153843/229785426-412577a1-7200-4bb7-a0fe-0cf086f6ee2d.png">

**Comment by Umesh2214906** [comment, marked as answer]
Thank You So Much Sir My Problem is Resolved it is working as expected

Thank you so much.
